pxr/imaging/hgiMetal/shaderGenerator.mm

```diff
@@ -303,9 +303,9 @@ void _Init(
         " atomic_fetch_add_explicit(&a, v, memory_order_relaxed)\n"
         "#define ATOMIC_EXCHANGE(a, desired)"
         " atomic_exchange_explicit(&a, desired, memory_order_relaxed)\n"
-        "#define ATOMIC_COMP_SWAP(a, expected, desired) \\"
-        "  atomic_compare_exchange_strong_explicit(&a, &expected, desired, "
-        "memory_order_relaxed, memory_order_relaxed)\n"
+        "#define ATOMIC_COMP_SWAP(a, expected, desired)"
+        "  atomic_compare_exchange_weak_explicit(&a, &expected, desired, "
+        "memory_order_relaxed, memory_order_relaxed) ? expected : expected;\n"
```

Comment:
Is `x() ? expected : expected` a typo? If the intention is to always return `expected` then I think `(x(), expected)` might work as well and be clearer.